aws_signing_helper/update.go

```diff
@@ -98,90 +98,92 @@ func GetCredentialsFileContents() ([]string, error) {
 }
 
 // Assume that the credentials file exists already and open it for write operations
+// that will overwrite the existing contents of the file
 func GetWriteOnlyCredentialsFile() (*os.File, error) {
 	homeDir, _ := os.UserHomeDir()
 	awsCredentialsPath := os.Getenv(AwsSharedCredentialsFileEnvVarName)
 	if awsCredentialsPath == "" {
 		awsCredentialsPath = filepath.Join(homeDir, ".aws", "credentials")
 	}
-	return os.OpenFile(awsCredentialsPath, os.O_WRONLY, 0200)
+	return os.OpenFile(awsCredentialsPath, os.O_WRONLY|os.O_TRUNC, 0200)
 }
 
 // Function to write existing credentials and newly-created credentials to a destination file
-func WriteTo(profileName string, writeLines []string, cred *TemporaryCredential) error {
+func WriteTo(profileName string, readLines []string, cred *TemporaryCredential) error {
 	destFile, err := GetWriteOnlyCredentialsFile()
 	if err != nil {
 		log.Println("unable to get write-only AWS credentials file")
 		os.Exit(1)
 	}
 	defer destFile.Close()
 
-	// Create buffered writer with a buffer of maximum size
-	destFileWriter := bufio.NewWriterSize(destFile, math.MaxInt32)
-
 	var profileExist = false
 	var profileSection = "[" + profileName + "]"
```

Comment:
Is it legal to have any space between the delimiters and the actual profile name?

Reply:
No, it's illegal.